### DISCUSSION TITLE
How can I apply "initial_new_capacity_up" and "growth_rate" for each mode?

### DISCUSSION THREAD

**Discussion by lcw95** [iiasa/message_ix#760] [Q&A] [answered]
Hello! Thank you for your efforts in MESSAGEix model development and for helping clarify my questions.

I have one question, which is why I'm writing to you. I'm looking to apply retrofit to tec1 technology. In this case, I'd like to utilize the mode functionality rather than add-ons. **_Is it possible to set "initial_new_capacity_up/lo" and "growth_rate" for each mode?_**

For instance, let's assume there are **tec1** technologies reflecting efficiency improvement called **tec1_eff** and fuel conversion called **_tec1_ctw_**. Tec1 has the **_"standard" mode_**, tec1_eff has **_"mit_1" mode_**, and tec1_ctw has **_"mit_2" mode._**

In reality, tec1 should naturally transition from tec1 to tec1_eff and then to tec1_ctw. However, due to a significant price competitiveness difference between the two technologies, the transition is taking place directly from tec1 to tec1_ctw. Additionally, there's a problem where tec1's ACT suddenly becomes 0 and shifts to tec1_ctw (corner solution).

Therefore, I would like to set  "initial_new_capacity_up/lo" and "growth_rate" for each mode. **However, it seems that "Equation NEW_CAPACITY_CONSTRAINT_UP" is not set up to apply to modes**.

**_Have you experienced this issue before or do you have any other solutions to share?_** 

Thank you for reading this long message.

----------------------------------
**_Summary:_**
1. Implementing the retrofit function using the mode feature.
2. Due to issues like corner solutions, I want to implement "initial_new_capacity_up/lo" and "growth_rate" functions for each mode.
3. Since the related equations do not include mode variables, I'm curious if there are alternative solutions.
----------------------------------

**Comment by lcw95** [comment, marked as answer]
I think I can easily solve this problem using "Share constraints on activity by mode"... I want to remove my discussion :(

**Reply by glatterf42** [reply]
Are you sure you want to delete this discussion? I can do that for you, but you might want to confirm that your solution works before we do that.

**Reply by lcw95** [reply]
It was a hasty decision to ask you to delete it! 

I can use this equation to implement it in the way I want. But I think it would be best to have the "initial_new_capacity_up/lo" and "growth_rate" functions reflected per mode. If you have any experience with this, please share me!

**Reply by glatterf42** [reply]
I've asked the team if they have experience with it or have been working on including a `mode` dimension with `initial_new_capacity_up`, but no one has. The issue is that `mode` defines _how_ a technology is operating, while parameters like `initial_new_capacity_up` define _how much_ of it is operating. So these capacity-related parameters will likely not reflect activity-related dimension. In our [documentation](https://docs.messageix.org/en/latest/model/MESSAGE/parameter_def.html#dynamic-constraints-on-new-capacity-and-activity), you can read that "these parameters do not have a mode () dimension, they cannot be used to constraint the activity/new capacity of single modes of technologies; only the total across all modes".

For setting analogous limits for the modes, you can indeed use the [equation](https://docs.messageix.org/en/latest/model/MESSAGE/model_core.html#share-constraints-on-activity-by-mode) you identified earlier.

If the above does not satisfy you, please explain again (and maybe in a new discussion) what you are looking for.
